pkg/kubelet/server/auth.go

```diff
@@ -81,35 +83,60 @@ func (n nodeAuthorizerAttributesGetter) GetRequestAttributes(u user.Info, r *htt
 
 	requestPath := r.URL.Path
 
-	// Default attributes mirror the API attributes that would allow this access to the kubelet API
-	attrs := authorizer.AttributesRecord{
-		User:            u,
-		Verb:            apiVerb,
-		Namespace:       "",
-		APIGroup:        "",
-		APIVersion:      "v1",
-		Resource:        "nodes",
-		Subresource:     "proxy",
-		Name:            string(n.nodeName),
-		ResourceRequest: true,
-		Path:            requestPath,
-	}
-
-	// Override subresource for specific paths
-	// This allows subdividing access to the kubelet API
+	fineGrained := utilfeature.DefaultFeatureGate.Enabled(features.KubeletFineGrainedAuthz)
+	var attrs []authorizer.Attributes
```

Comment:
optional: I think it's just the subresource that changes between the cases below. It might be a little cleaner to just make this `var subresources []string`, and then loop over it at the end to construct the attributes.

Reply:
Done.